dist/mapdc.js

```diff
@@ -26631,21 +26629,24 @@ function rasterLayerPolyMixin(_layer) {
         layerFilter = _ref$layerFilter === undefined ? [] : _ref$layerFilter,
         filtersInverse = _ref.filtersInverse;
 
-    var selfJoin = state.data[0].table === state.data[1].table;
+    var doJoin = state.data.length > 1;
+    var hasColorMeasure = !(state.encoding.color.domain === undefined);
 
-    var groupby = {
+    var groupby = hasColorMeasure && !(state.data[0].attr === "rowid") ? {
```

Comment:
Maybe `!(state.data[0].attr === "rowid")` could go in a variable to give some info about what it means